src/coreclr/jit/forwardsub.cpp

```diff
@@ -274,6 +266,20 @@ class ForwardSubVisitor final : public GenTreeVisitor<ForwardSubVisitor>
         }
 
         m_accumulatedFlags |= (node->gtFlags & GTF_GLOB_EFFECT);
+        if ((node->gtFlags & GTF_CALL) != 0)
```

Comment:
This was intentional when I first added this API. It's really because of what @markples mentions below: `OperExceptions` does not make sense for user calls. The return value of `ExceptionSetFlags::All` indicates that something can throw all the built-in exceptions, but user calls can throw many more exceptions types. So to model that we would probably add some `ExceptionSetFlags::UnknownException`, but then the callers would need to either handle it specially or assert that they didn't see it.
The handling here by setting `All` is really just relying on forward-sub specific knowledge that the caller is going to handle that in the correct way, it's not actually correct to say that `GTF_CALL` nodes can throw those exceptions only.

> I wonder if there's any gain to be had by allowing some helper calls to forward sub... probably not much.

Definitely something we could try. We would need to generalize things a bit here to track more precise effects, since at that point we are throwing away the "`GTF_CALL` means user call" assumption that the current interference checks in the caller are relying on. (But that might also make sense if we are going to end up with a `GT_BOX` that has `GTF_CALL` set on it in the future.)